.github/dependabot.yml

```diff
@@ -0,0 +1,38 @@
+version: 2
+updates:
+  - package-ecosystem: bundler
+    directory: /
+    schedule:
+      interval: daily
+    allow:
+      # Security updates
+      - dependency-name: brakeman
+        dependency-type: direct
+      # Internal gems
+      - dependency-name: "govuk*"
+        dependency-type: direct
+      - dependency-name: "rubocop-govuk"
+        dependency-type: direct
+      - dependency-name: "gds-sso"
+        dependency-type: direct
+      - dependency-name: govspeak
+        dependency-type: direct
+      - dependency-name: plek
+        dependency-type: direct
+      - dependency-name: pg
+        dependency-type: direct
+      # Framework gems
+      - dependency-name: factory_bot_rails
+        dependency-type: direct
+      - dependency-name: rails
+        dependency-type: direct
+      - dependency-name: rspec
+        dependency-type: direct
+      - dependency-name: rspec-rails
+        dependency-type: direct
+  - package-ecosystem: docker
```

Comment:
What does this do? I hadn't noticed it before.

Reply:
This has Dependabot checking Dockerfile for dependencies. There's not a lot of info about it on the [GitHub native Dependabot docs](https://docs.github.com/en/github/administering-a-repository/enabling-and-disabling-version-updates), but the [release notes from 2017](https://dependabot.com/blog/dependabot-now-supports-docker/) suggest that it looks at the `FROM` field in the Dockerfile and checks for newer versions.

I've been going on the basis that if a repo has a Dockerfile, it can't hurt to have a corresponding check in Dependabot. However, we're explicitly ignoring updates to `ruby` as upgrading ruby requires manual intervention, so it's quite possible that this extra config is giving us nothing. I could omit the Docker config altogether...?

Reply:
I preferred the idea of leaving it in to make it more explicit (if it's not there, someone might think one day we could enable Docker upgrades only to discover that it doesn't really work as we have the Ruby version stored in multiple places).

I was also thinking that in the future there might be some other way we use Docker and it might be nice having a pre-configured configuration file... but that's just anticipation...

Reply:
Oh I see, so we're actively turning this on in order to turn it off? ie, we expect the only dependency it'll manage is ruby (as it's the only thing with a FROM) and so we're telling it not to do that?

Am i understanding correctly there? if so do we need the config for directory and schedule?

Reply:
That's correct. And yes, I think [directory and schedule are required fields](https://docs.github.com/en/github/administering-a-repository/enabling-and-disabling-version-updates#enabling-github-dependabot-version-updates).

Reply:
Sure, could we add a comment to explain at all? it feels like rather niche information.

Reply:
Done 👍 could you check it makes sense?